src/httpClient/default.ts

```diff
@@ -106,7 +95,7 @@ export default class DefaultHTTPClient implements HTTPClient {
       error = null;
       request_start = performance.now();
       try {
-        response = await fetch(new URL(path, this.origin), {
+        response = await fetch(new URL(updatedPath, this.origin), {
```

Comment:
I think this might be problematic, especially if the origin contains a path segment

Reply:
i like the idea of just building the url instead of getting the updated path and then awkwardly trying to fit that into a `new URL()`!